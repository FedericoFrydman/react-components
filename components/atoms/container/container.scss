@import '../../assets/styles/colors.scss';

.lsux-container {

  &.lsux-container--light-gray {
    background-color: $N100;
  }
  &.lsux-container--white {
    background-color: $N00;
  }
  &.lsux-container--fixed {
    position: fixed;
    z-index: 99;
  }
  &.lsux-container--flexbox {
    display: flex;

    &.lsux-container--flex-direction-row {
      flex-direction: row;
    }
    &.lsux-container--flex-direction-row-reverse {
      flex-direction: row-reverse;
    }
    &.lsux-container--flex-direction-column {
      flex-direction: column;
    }
    &.lsux-container--flex-direction-column-reverse {
      flex-direction: column-reverse;
    }

    &.lsux-container--flex-wrap-wrap {
      flex-wrap: wrap;
    }
    &.lsux-container--flex-wrap-nowrap {
      flex-wrap: nowrap;
    }
    &.lsux-container--flex-wrap-reverse {
      flex-wrap: wrap-reverse;
    }

    &.lsux-container--flex-justify-flex-start {
      justify-content: flex-start;
    }
    &.lsux-container--flex-justify-flex-end {
      justify-content: flex-end;
    }
    &.lsux-container--flex-justify-center {
      justify-content: center;
    }
    &.lsux-container--flex-justify-space-between {
      justify-content: space-between;
    }
    &.lsux-container--flex-justify-space-around {
      justify-content: space-around;
    }

    &.lsux-container--flex-items-flex-start {
      align-items: flex-start;
    }
    &.lsux-container--flex-items-flex-end {
      align-items: flex-end;
    }
    &.lsux-container--flex-items-center {
      align-items: center;
    }
    &.lsux-container--flex-items-baseline {
      align-items: baseline;
    }
    &.lsux-container--flex-items-stretch {
      align-items: stretch;
    }

    &.lsux-container--flex-content-flex-start {
      align-content: flex-start;
    }
    &.lsux-container--flex-content-flex-end {
      align-content: flex-end;
    }
    &.lsux-container--flex-content-center {
      align-content: center;
    }
    &.lsux-container--flex-content-space-between {
      align-content: space-between;
    }
    &.lsux-container--flex-content-space-around {
      align-content: space-around;
    }
    &.lsux-container--flex-content-strech {
      align-content: stretch;
    }
  }
}
