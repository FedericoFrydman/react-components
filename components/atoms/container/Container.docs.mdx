import { ArgsTable, Story, Canvas, Stories } from '@storybook/addon-docs';
import { Alert } from '../../molecules/alert/Alert';
import { Container } from './Container';

# Container

<Canvas>
  <Story id="layout-container--light-gray-background" />
</Canvas>

<Alert title="Dev Note" dismissible={false} severity="warning" icon="alert_warning">
  This component is just to use it when the child container has special specifications, like background, position,
  etc...
</Alert>

<p style={{ margin: 0 }}>&nbsp;</p>

<ArgsTable of={Container}/>

## Example Code

```tsx
<Container background="light-gray" fixed={true}>
  // the children go here
<Container/>
```

## Usage Guideline

| Prop       | Usage                                                                                                                                                             |
| ---------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| background | This prop should be equal to "light-gray" in order to add a class to the container that will provied the background color. Now it only support light-gray ($N100) |
| fixed      | When it is true, it adds a class to the container to make it position fixed with 99 z-index                                                                       |
| flexbox    | When it is true, it adds a class to the container to make it display flex                                                                                         |
