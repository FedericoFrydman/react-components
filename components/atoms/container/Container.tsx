import { forwardRef, Ref, ReactElement } from 'react';
import cn from 'classnames';

import './container.scss';

export type ContainerBgColors = 'light-gray' | 'white' | 'none';

export interface ContainerProps extends Omit<React.HTMLProps<HTMLDivElement>, 'css'> {
  /**
   * Define if container has background
   * @default 'white'
   */
  background?: ContainerBgColors;
  /**
   * Define if container has position fixed
   */
  fixed?: boolean;
  /**
   * Define if container has display flex
   */
  flexbox?: boolean;
  /**
   * Define the flex-direction of flex-container
   */
  flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
  /**
   * Define the flex-wrap of flex-container
   */
  flexWrap?: 'wrap' | 'nowrap' | 'wrap-reverse';
  /**
   * Define the flex-direction
   */
  justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around';
  /**
   * Define the align-items for flex-container
   */
  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
  /**
   * Define the flex-direction
   */
  alignContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'stretch';
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * Children Container
   */
  children?: string | number | ReactElement | ReactElement[];
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLDivElement>;
}

const ContainerComponent = (
  {
    background = 'white',
    classNames = [],
    children,
    fixed,
    flexbox,
    flexDirection,
    flexWrap,
    justifyContent,
    alignItems,
    alignContent,
    ...props
  }: ContainerProps,
  ref: ContainerProps['ref'],
) => {
  const classContainer = 'lsux-container';
  const cnContainer = cn(
    {
      [`${classContainer}`]: true,
      [`${classContainer}--${background}`]: background !== 'none',
      [`${classContainer}--fixed`]: fixed,
      [`${classContainer}--flexbox`]: flexbox,
      [`${classContainer}--flex-direction-${flexDirection}`]: flexDirection,
      [`${classContainer}--flex-wrap-${flexWrap}`]: flexWrap,
      [`${classContainer}--flex-justify-${justifyContent}`]: justifyContent,
      [`${classContainer}--flex-items-${alignItems}`]: alignItems,
      [`${classContainer}--flex-content-${alignContent}`]: alignContent,
    },
    classNames,
  );

  return (
    <div {...props} className={cnContainer} ref={ref}>
      {children}
    </div>
  );
};

export const Container = forwardRef<HTMLDivElement, ContainerProps>(ContainerComponent);
