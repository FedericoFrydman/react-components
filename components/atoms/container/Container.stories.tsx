import { Story, Meta } from '@storybook/react';
import '../../assets/styles/utilities.scss';

import docs from './Container.docs.mdx';
import { Container, ContainerProps } from './Container';
import { Text } from '../text/Text';

const ContainerStory: Story<ContainerProps> = (args: ContainerProps) => (
  <Container {...args}>
    <Text text="I am container child!" as="p" classNames={['p-5']} />
  </Container>
);

export const WhiteBackground = ContainerStory.bind({});
WhiteBackground.args = {
  background: 'white',
};

export const LightGrayBackground = ContainerStory.bind({});
LightGrayBackground.args = {
  background: 'light-gray',
};

export default {
  argTypes: {
    background: {
      options: ['light-gray', 'white', 'none'],
      type: 'select',
    },
  },
  component: ContainerStory,
  parameters: {
    docs: { page: docs },
  },
  title: 'Layout/Container',
} as Meta;
