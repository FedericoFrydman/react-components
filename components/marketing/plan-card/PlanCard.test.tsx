import { createRef } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { render, screen } from '@testing-library/react';

import {
  PlanCard,
  IPlanCardProps,
  PlanCardInfo,
  PlanCardAddOns,
  PlanCardRegionPicker,
  planCardAvailableRegions,
  IPlanCardRegionPickerProps,
  IPlanCardInfoProps,
} from './PlanCard';
import { ContainerBgColors } from '../../../components/atoms/container/Container';

describe('PlanCard Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PlanCard id="ls-card" isSelected={true} selectable>
        <PlanCardInfo
          appName="LegalShield"
          planDescription={'Legal protection for you and your family'}
          planFees={'+ processing fees'}
          planName={'Individual & Family Legal'}
          planPrice={'$29.95'}
          planRecurrence={'/mo'}
          planType={'Legal Plan'}
          showFees={true}
          onClick={null}
          onSeeDetailsClick={() => alert('See more!')}
        >
          <PlanCardRegionPicker
            availableRegions={planCardAvailableRegions}
            selectedRegion={'CA'}
            onRegionSelect={null}
          />
        </PlanCardInfo>
        <PlanCardAddOns />
      </PlanCard>,
    );
    testId = 'test plan card';
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the card the provided classNames', () => {
          const card = renderPlanCard({ classNames });

          expect(card).toHaveClass(classNames.join(' '), { exact: false });
        });
      });
    });

    describe('isSelected', () => {
      const selectedClassName = ['lsux-plan-card--selected'].join(' ');

      it('gives the card the selected class', () => {
        const card = renderPlanCard({ isSelected: true });

        expect(card).toHaveClass(selectedClassName, { exact: false });
      });
    });

    describe('background', () => {
      const backgrounds: ContainerBgColors[] = ['light-gray', 'white', 'none'];

      it('defaults to white', () => {
        const card = renderPlanCard();

        expect(card).toHaveClass(`lsux-card--bg--white`);
      });

      function itAssignsTheRightClassForBackground(background: ContainerBgColors) {
        const cardBgClass = `lsux-card--bg--${background}`;

        describe(`when the background is ${background}`, () => {
          it(`assigns the ${cardBgClass} to the card`, () => {
            const card = renderPlanCard({ background: background });

            expect(card).toHaveClass(cardBgClass);
          });
        });
      }

      backgrounds.forEach((bg) => itAssignsTheRightClassForBackground(bg));
    });

    describe('cardHeight', () => {
      describe('when "cardHeight" is provided', () => {
        const cardHeight = `200px`;

        it('sets the card width to the provided value', () => {
          const card = renderPlanCard({ cardHeight });

          expect(card).toHaveStyle({ height: cardHeight });
        });
      });
    });

    describe('cardWidth', () => {
      describe('when "cardWidth" is provided', () => {
        const cardWidth = `200px`;

        it('sets the card width to the provided value', () => {
          const card = renderPlanCard({ cardWidth });

          expect(card).toHaveStyle({ width: cardWidth });
        });
      });
    });
  });

  describe('ref', () => {
    const ref = createRef<HTMLDivElement>();

    it('provides a ref to the card', () => {
      expect(renderPlanCard({ ref })).toEqual(ref.current);
    });
  });

  describe('unspecified html attributes', () => {
    const altText = 'My accessible card';

    it('allows html attributes to be passed to the underlying card element', () => {
      render(<PlanCard alt={altText} data-testid={testId} />);

      const card = screen.getByTestId(testId);

      expect(card).toHaveAttribute('alt', altText);
    });
  });

  const renderPlanCard = (props: IPlanCardProps = {}): HTMLElement => {
    render(<PlanCard {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});

describe('PlanCardInfo Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PlanCardInfo
        appName="LegalShield"
        planDescription={'Legal protection for you and your family'}
        planFees={'+ processing fees'}
        planName={'Individual & Family Legal'}
        planPrice={'$29.95'}
        planRecurrence={'/mo'}
        planType={'Legal Plan'}
        showFees={true}
        onClick={null}
        onSeeDetailsClick={() => alert('See more!')}
      />,
    );
    testId = 'test-plan-card-info';
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('planDescription', () => {
      describe('when provided', () => {
        const planDescription = 'Test plan description';

        it('renders the plan description', () => {
          const cardInfo = renderPlanCardInfo({ planDescription });

          expect(cardInfo).toHaveTextContent(planDescription);
        });
      });
    });
  });

  const renderPlanCardInfo = (props: IPlanCardInfoProps = {}): HTMLElement => {
    render(<PlanCardInfo {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});

describe('PlanCardRegionPicker Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PlanCardRegionPicker
        availableRegions={planCardAvailableRegions}
        selectedRegion={'CA'}
        onRegionSelect={null}
        invalidRegions={[]}
        invalidRegionComponent={null}
        invalidRegionMessage="Test invalid region message"
        showValidationMessage={true}
      />,
    );
    testId = 'test-plan-card-region-picker';
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('availableRegions', () => {
      it('renders the provided regions', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('California');
      });
    });
    describe('invalidRegions', () => {
      it('renders the region even when invalid', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          invalidRegions: ['CA'],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('California');
      });
    });
    describe('invalidRegionComponent', () => {
      it('renders the provided component when invalid', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          invalidRegionComponent: <div>You canot purchase this plan in your state.</div>,
          invalidRegions: ['CA'],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('You canot purchase this plan in your state.');
      });
    });
    describe('invalidRegionMessage', () => {
      it('renders the provided message when invalid', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          invalidRegionMessage: 'Please call us for more information.',
          invalidRegions: ['CA'],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('Please call us for more information.');
      });
    });
    describe('invalidRegionTitle', () => {
      it('renders the provided title when invalid', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          invalidRegionTitle: 'This plan is not available in your state.',
          invalidRegions: ['CA'],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('This plan is not available in your state.');
      });
    });
    describe('selectedRegion', () => {
      it('renders the provided selected region', () => {
        const regionPicker = renderPlanCardRegionPicker({
          availableRegions: [{ label: 'California', value: 'CA' }],
          selectedRegion: 'CA',
        });

        expect(regionPicker).toHaveTextContent('California');
      });
    });
    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the card the provided classNames', () => {
          const card = renderPlanCardRegionPicker({ classNames });

          expect(card).toHaveClass(classNames.join(' '), { exact: false });
        });
      });
    });
  });

  const renderPlanCardRegionPicker = (props: IPlanCardRegionPickerProps = {}): HTMLElement => {
    render(<PlanCardRegionPicker {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});
