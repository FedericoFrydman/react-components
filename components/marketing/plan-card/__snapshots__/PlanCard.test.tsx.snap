// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanCard Component should render the component 1`] = `
<Card
  classNames={
    Array [
      "lsux-plan-card lsux-plan-card--selected",
    ]
  }
  id="ls-card"
  isSelected={true}
  selectable={true}
>
  <PlanCardInfo
    appName="LegalShield"
    onClick={null}
    onSeeDetailsClick={[Function]}
    planDescription="Legal protection for you and your family"
    planFees="+ processing fees"
    planName="Individual & Family Legal"
    planPrice="$29.95"
    planRecurrence="/mo"
    planType="Legal Plan"
    showFees={true}
  >
    <PlanCardRegionPicker
      availableRegions={
        Array [
          Object {
            "label": "Alabama",
            "value": "AL",
          },
          Object {
            "label": "Alaska",
            "value": "AK",
          },
          Object {
            "label": "Arizona",
            "value": "AZ",
          },
          Object {
            "label": "Arkansas",
            "value": "AR",
          },
          Object {
            "label": "California",
            "value": "CA",
          },
          Object {
            "label": "Colorado",
            "value": "CO",
          },
          Object {
            "label": "Connecticut",
            "value": "CT",
          },
          Object {
            "label": "Delaware",
            "value": "DE",
          },
          Object {
            "label": "Florida",
            "value": "FL",
          },
          Object {
            "label": "Georgia",
            "value": "GA",
          },
          Object {
            "label": "Hawaii",
            "value": "HI",
          },
          Object {
            "label": "Idaho",
            "value": "ID",
          },
          Object {
            "label": "Illinois",
            "value": "IL",
          },
          Object {
            "label": "Indiana",
            "value": "IN",
          },
          Object {
            "label": "Iowa",
            "value": "IA",
          },
          Object {
            "label": "Kansas",
            "value": "KS",
          },
          Object {
            "label": "Kentucky",
            "value": "KY",
          },
          Object {
            "label": "Louisiana",
            "value": "LA",
          },
          Object {
            "label": "Maine",
            "value": "ME",
          },
          Object {
            "label": "Maryland",
            "value": "MD",
          },
          Object {
            "label": "Massachusetts",
            "value": "MA",
          },
          Object {
            "label": "Michigan",
            "value": "MI",
          },
          Object {
            "label": "Minnesota",
            "value": "MN",
          },
          Object {
            "label": "Mississippi",
            "value": "MS",
          },
          Object {
            "label": "Missouri",
            "value": "MO",
          },
          Object {
            "label": "Montana",
            "value": "MT",
          },
          Object {
            "label": "Nebraska",
            "value": "NE",
          },
          Object {
            "label": "Nevada",
            "value": "NV",
          },
          Object {
            "label": "New Hampshire",
            "value": "NH",
          },
          Object {
            "label": "New Jersey",
            "value": "NJ",
          },
          Object {
            "label": "New Mexico",
            "value": "NM",
          },
          Object {
            "label": "New York",
            "value": "NY",
          },
          Object {
            "label": "North Carolina",
            "value": "NC",
          },
          Object {
            "label": "North Dakota",
            "value": "ND",
          },
          Object {
            "label": "Ohio",
            "value": "OH",
          },
          Object {
            "label": "Oklahoma",
            "value": "OK",
          },
          Object {
            "label": "Oregon",
            "value": "OR",
          },
          Object {
            "label": "Pennsylvania",
            "value": "PA",
          },
          Object {
            "label": "Rhode Island",
            "value": "RI",
          },
          Object {
            "label": "South Carolina",
            "value": "SC",
          },
          Object {
            "label": "South Dakota",
            "value": "SD",
          },
          Object {
            "label": "Tennessee",
            "value": "TN",
          },
          Object {
            "label": "Texas",
            "value": "TX",
          },
          Object {
            "label": "Utah",
            "value": "UT",
          },
          Object {
            "label": "Vermont",
            "value": "VT",
          },
          Object {
            "label": "Virginia",
            "value": "VA",
          },
          Object {
            "label": "Washington",
            "value": "WA",
          },
          Object {
            "label": "West Virginia",
            "value": "WV",
          },
          Object {
            "label": "Wisconsin",
            "value": "WI",
          },
          Object {
            "label": "Wyoming",
            "value": "WY",
          },
        ]
      }
      onRegionSelect={null}
      selectedRegion="CA"
    />
  </PlanCardInfo>
  <PlanCardAddOns />
</Card>
`;

exports[`PlanCardInfo Component should render the component 1`] = `
<CardContent>
  <div
    className="lsux-plan-card__header"
    onClick={null}
  >
    <div
      className="lsux-plan-card__header__logo-container"
    >
      <ForwardRef(IconComponent)
        classNames="lsux-plan-card__header__logo-container__icon"
        name="apps_legal_shield"
        size="xlarge"
      />
    </div>
    <div
      className="lsux-plan-card__header__title-container"
    >
      <div>
        <ForwardRef(HeadingComponent)
          as="T16"
          classNames={
            Array [
              "pc-plan-name",
            ]
          }
          text="Individual & Family Legal"
        />
        <ForwardRef(TextComponent)
          classNames={
            Array [
              "pc-plan-type",
            ]
          }
          text="Legal Plan"
          textSize="small"
        />
      </div>
    </div>
  </div>
  <div
    className="lsux-plan-card__plan-info"
  >
    <div
      className="lsux-plan-card__plan-info-price pt-4"
    >
      <ForwardRef(TitleComponent)
        text="$29.95/mo"
        textSize="large"
      />
      <ForwardRef(TextComponent)
        classNames={
          Array [
            "plan-fee-text",
            "pt-1",
          ]
        }
        text="+ processing fees"
        textSize="small"
      />
    </div>
    <div
      className="lsux-plan-card__plan-info-description pt-4 pb-3"
    >
      <ForwardRef(TextComponent)
        text="Legal protection for you and your family"
      />
    </div>
  </div>
</CardContent>
`;

exports[`PlanCardRegionPicker Component should render the component 1`] = `
<div
  className="lsux-plan-card__region-selector pt-5"
>
  <ForwardRef(SelectComponent)
    onValueChange={null}
    placeholder="Select your state"
    stretch={true}
    value="CA"
  >
    <SelectItem
      key="AL"
      value="AL"
    >
      Alabama
    </SelectItem>
    <SelectItem
      key="AK"
      value="AK"
    >
      Alaska
    </SelectItem>
    <SelectItem
      key="AZ"
      value="AZ"
    >
      Arizona
    </SelectItem>
    <SelectItem
      key="AR"
      value="AR"
    >
      Arkansas
    </SelectItem>
    <SelectItem
      key="CA"
      value="CA"
    >
      California
    </SelectItem>
    <SelectItem
      key="CO"
      value="CO"
    >
      Colorado
    </SelectItem>
    <SelectItem
      key="CT"
      value="CT"
    >
      Connecticut
    </SelectItem>
    <SelectItem
      key="DE"
      value="DE"
    >
      Delaware
    </SelectItem>
    <SelectItem
      key="FL"
      value="FL"
    >
      Florida
    </SelectItem>
    <SelectItem
      key="GA"
      value="GA"
    >
      Georgia
    </SelectItem>
    <SelectItem
      key="HI"
      value="HI"
    >
      Hawaii
    </SelectItem>
    <SelectItem
      key="ID"
      value="ID"
    >
      Idaho
    </SelectItem>
    <SelectItem
      key="IL"
      value="IL"
    >
      Illinois
    </SelectItem>
    <SelectItem
      key="IN"
      value="IN"
    >
      Indiana
    </SelectItem>
    <SelectItem
      key="IA"
      value="IA"
    >
      Iowa
    </SelectItem>
    <SelectItem
      key="KS"
      value="KS"
    >
      Kansas
    </SelectItem>
    <SelectItem
      key="KY"
      value="KY"
    >
      Kentucky
    </SelectItem>
    <SelectItem
      key="LA"
      value="LA"
    >
      Louisiana
    </SelectItem>
    <SelectItem
      key="ME"
      value="ME"
    >
      Maine
    </SelectItem>
    <SelectItem
      key="MD"
      value="MD"
    >
      Maryland
    </SelectItem>
    <SelectItem
      key="MA"
      value="MA"
    >
      Massachusetts
    </SelectItem>
    <SelectItem
      key="MI"
      value="MI"
    >
      Michigan
    </SelectItem>
    <SelectItem
      key="MN"
      value="MN"
    >
      Minnesota
    </SelectItem>
    <SelectItem
      key="MS"
      value="MS"
    >
      Mississippi
    </SelectItem>
    <SelectItem
      key="MO"
      value="MO"
    >
      Missouri
    </SelectItem>
    <SelectItem
      key="MT"
      value="MT"
    >
      Montana
    </SelectItem>
    <SelectItem
      key="NE"
      value="NE"
    >
      Nebraska
    </SelectItem>
    <SelectItem
      key="NV"
      value="NV"
    >
      Nevada
    </SelectItem>
    <SelectItem
      key="NH"
      value="NH"
    >
      New Hampshire
    </SelectItem>
    <SelectItem
      key="NJ"
      value="NJ"
    >
      New Jersey
    </SelectItem>
    <SelectItem
      key="NM"
      value="NM"
    >
      New Mexico
    </SelectItem>
    <SelectItem
      key="NY"
      value="NY"
    >
      New York
    </SelectItem>
    <SelectItem
      key="NC"
      value="NC"
    >
      North Carolina
    </SelectItem>
    <SelectItem
      key="ND"
      value="ND"
    >
      North Dakota
    </SelectItem>
    <SelectItem
      key="OH"
      value="OH"
    >
      Ohio
    </SelectItem>
    <SelectItem
      key="OK"
      value="OK"
    >
      Oklahoma
    </SelectItem>
    <SelectItem
      key="OR"
      value="OR"
    >
      Oregon
    </SelectItem>
    <SelectItem
      key="PA"
      value="PA"
    >
      Pennsylvania
    </SelectItem>
    <SelectItem
      key="RI"
      value="RI"
    >
      Rhode Island
    </SelectItem>
    <SelectItem
      key="SC"
      value="SC"
    >
      South Carolina
    </SelectItem>
    <SelectItem
      key="SD"
      value="SD"
    >
      South Dakota
    </SelectItem>
    <SelectItem
      key="TN"
      value="TN"
    >
      Tennessee
    </SelectItem>
    <SelectItem
      key="TX"
      value="TX"
    >
      Texas
    </SelectItem>
    <SelectItem
      key="UT"
      value="UT"
    >
      Utah
    </SelectItem>
    <SelectItem
      key="VT"
      value="VT"
    >
      Vermont
    </SelectItem>
    <SelectItem
      key="VA"
      value="VA"
    >
      Virginia
    </SelectItem>
    <SelectItem
      key="WA"
      value="WA"
    >
      Washington
    </SelectItem>
    <SelectItem
      key="WV"
      value="WV"
    >
      West Virginia
    </SelectItem>
    <SelectItem
      key="WI"
      value="WI"
    >
      Wisconsin
    </SelectItem>
    <SelectItem
      key="WY"
      value="WY"
    >
      Wyoming
    </SelectItem>
  </ForwardRef(SelectComponent)>
</div>
`;
